packages/vaadin-grid/test/array-data-provider.test.js

```diff
@@ -250,3 +251,40 @@ describe('invalid paths', () => {
     });
   });
 });
+
+describe('items with a custom data provider', () => {
+  let grid;
+  const dataProvider = (params, callback) => callback([{ value: 'foo' }], 1);
+  const items = [{ value: 'bar' }];
+
+  beforeEach(() => {
+    grid = fixtureSync(`
+      <vaadin-grid>
+        <vaadin-grid-column path="value"></vaadin-grid-column>
+      </vaadin-grid>
+    `);
+  });
+
+  it('use the items array', () => {
+    grid.dataProvider = dataProvider;
+    grid.items = items;
+    flushGrid(grid);
+    expect(getBodyCellContent(grid, 0, 0).textContent).to.equal('bar');
+    expect(grid.dataProvider).not.to.equal(dataProvider);
```

Comment:
Probably, this assertion is not necessary as there is [the relevant test](https://github.com/vaadin/web-components/pull/1976/files#diff-25fcc44d6a81f4b9e6b7626f9b8f28873c65581af1b52f4ddbde64378b4adefaR93) above.

Reply:
Done